src/Symfony/Component/Security/Http/Firewall/UsernamePasswordJsonAuthenticationListener.php

```diff
@@ -180,4 +180,11 @@ private function onFailure(Request $request, AuthenticationException $failed)
 
         return $response;
     }
+
+    private function throwBadRequestHttpException(Request $request, $message, \Exception $previous = null)
```

Comment:
I would not throw inside the method, as it means you have to inspect this method to know that the flow stops in the main method. Instead, create a method *returning* the exception, and keep throwing it in the main method